src/ext.rs

```diff
@@ -1,37 +1,37 @@
 use super::*;
 use std::ops::Deref;
 
-impl<T, O, I, F> Traversal for Map<T, O, I, F>
-where I: Traversal<Item=T>, F: FnMut(T) -> O {
-    type Item = O;
+impl<I, F> Traversal for Map<I, F>
+where I: Traversal, F: FnMut<(<I as Traversal>::Item,)> {
```

Comment:
I learned that if you put the `I: Traversal` bound in the `impl` section, `I::Item` will Just Work.